pkg/monitortestlibrary/pathologicaleventlibrary/duplicated_event_patterns.go

```diff
@@ -2,419 +2,668 @@ package pathologicaleventlibrary
 
-	// Separated out in testMarketplaceStartupProbeFailure
-	regexp.MustCompile(MarketplaceStartupProbeFailureRegExpStr),
+	// Start with the main list of matchers:
+	registry := NewUniversalPathologicalEventMatchers(kubeConfig, finalIntervals)
 
-	// stale condition challenge reset
-	regexp.MustCompile(`message changed from "\x{FEFF}`),
-}
+	// Now add in the matchers we only want to apply during upgrade:
 
-// AllowedUpgradeRepeatedEventPatterns are patterns of events that we should only allow during upgrades, not during normal execution.
-var AllowedUpgradeRepeatedEventPatterns = []*regexp.Regexp{
 	// Operators that use library-go can report about multiple versions during upgrades.
-	regexp.MustCompile(`ns/openshift-etcd-operator deployment/etcd-operator - reason/MultipleVersions multiple versions found, probably in transition: .*`),
-	regexp.MustCompile(`ns/openshift-kube-apiserver-operator deployment/kube-apiserver-operator - reason/MultipleVersions multiple versions found, probably in transition: .*`),
-	regexp.MustCompile(`ns/openshift-kube-controller-manager-operator deployment/kube-controller-manager-operator - reason/MultipleVersions multiple versions found, probably in transition: .*`),
-	regexp.MustCompile(`ns/openshift-kube-scheduler-operator deployment/openshift-kube-scheduler-operator - reason/MultipleVersions multiple versions found, probably in transition: .*`),
+	registry.AddPathologicalEventMatcherOrDie(&SimplePathologicalEventMatcher{
+		name: "OperatorMultipleVersions",
+		locatorKeyRegexes: map[monitorapi.LocatorKey]*regexp.Regexp{
+			monitorapi.LocatorNamespaceKey:  regexp.MustCompile(`(openshift-etcd-operator|openshift-kube-apiserver-operator|openshift-kube-controller-manager-operator|openshift-kube-scheduler-operator)`),
+			monitorapi.LocatorDeploymentKey: regexp.MustCompile(`(etcd-operator|kube-apiserver-operator|kube-controller-manager-operator|openshift-kube-scheduler-operator)`),
+		},
+		messageReasonRegex: regexp.MustCompile(`^MultipleVersions$`),
+		messageHumanRegex:  regexp.MustCompile(`multiple versions found, probably in transition`),
+	})
 
 	// etcd-quorum-guard can fail during upgrades.
-	regexp.MustCompile(`ns/openshift-etcd pod/etcd-quorum-guard-[a-z0-9-]+ node/[a-z0-9.-]+ - reason/Unhealthy Readiness probe failed: `),
-	// etcd can have unhealthy members during an upgrade
-	regexp.MustCompile(`ns/openshift-etcd-operator deployment/etcd-operator - reason/UnhealthyEtcdMember unhealthy members: .*`),
-	// etcd-operator began to version etcd-endpoints configmap in 4.10 as part of static-pod-resource. During upgrade existing revisions will not contain the resource.
-	// The condition reconciles with the next revision which the result of the upgrade. TODO(hexfusion) remove in 4.11
-	regexp.MustCompile(`ns/openshift-etcd-operator deployment/etcd-operator - reason/RequiredInstallerResourcesMissing configmaps: etcd-endpoints-[0-9]+`),
```

Comment:
I don't see where this one is recovered.  The only one I see is https://github.com/dgoodwin/origin/blob/58516b3ec699c1490d114c7580eb0a951edc3c7c/pkg/monitortestlibrary/pathologicaleventlibrary/duplicated_event_patterns.go#L550-L555 but it doesn't have `LocatorDeploymentKey...etcd-operator`.

Reply:
ah, ignore my comment above.

Reply:
Yup you found it :) Now I just checked bigquery, and it's still happening. :)

```sql
SELECT prowjob_name, prowjob_build_id, from_time, payload.locator, payload.message FROM `openshift-gce-devel.ci_analysis_us.job_intervals` WHERE from_time >= DATETIME_SUB(CURRENT_DATETIME(), INTERVAL 14 DAY) 
AND message like '%pathological%reason/RequiredInstallerResourcesMissing configmaps: etcd%' 
AND locator like '%openshift-etcd%'
LIMIT 1000
```

This returned 71 hits in the last two weeks. That's not a huge hit rate, but it will mean a few failed jobs. Then again, those hits are all 4.14, none in 4.15. I say lets keep it out and move on.